check_logs.py

```diff
@@ -13,7 +14,43 @@
 def _fetch(title, url, dest):
     print_yellow("fetching %s from: %s" % (title, url))
     # TODO: use something more robust like python wget library.
-    urllib.request.urlretrieve(url, dest)
+    retries = 0
+    max_retries = 7
+    retry_codes = [500, 504]
+    while retries < max_retries:
+        try:
+            if retries:
+                time.sleep(2**retries)
+            retries += 1
+            urllib.request.urlretrieve(url, dest)
+            break
+        except ConnectionResetError as err:
+            print_yellow('%s download error ("%s"), retrying...' %
+                         (title, str(err)))
+            pass
+        except HTTPError as err:
```

Comment:
I guess this isn't the right way to refer to this class of error in python: https://github.com/ClangBuiltLinux/continuous-integration2/runs/3767136408?check_suite_focus=true

> NameError: name 'HTTPError' is not defined

I think this needs to be `urllib.error.HTTPError`.